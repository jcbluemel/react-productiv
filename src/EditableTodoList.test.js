import React from "react";
import { render, fireEvent } from "@testing-library/react";
import EditableTodoList from "./EditableTodoList";

const TEST_TODO_LIST = [
    {
        id: "1",
        title: "Test1",
        description: "test1",
        priority: 1
    },
    {
        id: "2",
        title: "Test2",
        description: "test2",
        priority: 2
    }
];

const mock = () => { };

it("renders without crashing", function () {
    render(
        <EditableTodoList
            todos={TEST_TODO_LIST}
            update={mock}
            remove={mock}
        />
    );
});

it("matches snapshot", function () {
    const { asFragment } = render(
        <EditableTodoList
            todos={TEST_TODO_LIST}
            update={mock}
            remove={mock}
        />
    );
    expect(asFragment()).toMatchSnapshot();
});

it("contains todo data", function () {
    const { container, debug } = render(
        <EditableTodoList
            todos={TEST_TODO_LIST}
            update={mock}
            remove={mock}
        />
    ); 
    
    expect(container.querySelectorAll(".Todo")).toHaveLength(2);
});