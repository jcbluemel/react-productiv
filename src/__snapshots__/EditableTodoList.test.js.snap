// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
          id="1"
        >
          <div>
            <b>
              Test1
            </b>
             
            <small>
              Priority: 1
            </small>
          </div>
          <div>
            <small>
              test1
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
          id="2"
        >
          <div>
            <b>
              Test2
            </b>
             
            <small>
              Priority: 2
            </small>
          </div>
          <div>
            <small>
              test2
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
